### PR TITLE
Bug 1349170 - [Regression] SecRandomCopyBytes() doesn't execute withi…

### PR DESCRIPTION
**Pull request by justindarc**
…n an assert()

### CODE REVIEW

File: Shared/Bytes.swift [hunk 1]
```diff
@@ -11,8 +11,13 @@ public typealias GUID = String
  */
 open class Bytes {
     open class func generateRandomBytes(_ len: UInt) -> Data {
-        let bytes = [UInt32](repeating: 0, count: Int(len)).map { _ in arc4random() }
-        return Data(bytes: bytes, count: Int(len))
+        let len = Int(len)
+        var data = Data(count: len)
+        data.withUnsafeMutableBytes { (p: UnsafeMutablePointer<UInt8>) in
+            let result = SecRandomCopyBytes(kSecRandomDefault, len, p)
+            assert(result == 0, "Random byte generation failed.")
```
Comment (jvehent) [inline Shared/Bytes.swift]: Wasn't the original issue caused by assert being disabled in optimized builds? If so, this test would be disabled should optimized build be used again.

Comment (justindarc) [inline Shared/Bytes.swift]: This moves the computation outside of the assertion as was recommended here: https://bugs.swift.org/browse/SR-4308

Comment (jvehent) [inline Shared/Bytes.swift]: I'm not sure how other clients do it (@rnewman might know), but I imagine being 100% certain the quality test happens is important.

Comment (rnewman) [inline Shared/Bytes.swift]: I would be quite happy to have the client crash hard if `SRCB` returns anything other than `errSecSuccess`. We shouldn't proceed with whatever needed the value, and that means creating databases, computing GUIDs, or syncing. Crashing is our main way of finding out that there's a problem.

If we're building with normal optimizations, then switch `assert` for `precondition`.

If we're building with `-Ounchecked`, then you'll need to manually implement precondition using `fatalError`:

```swift
if (result != 0) {
    fatalError("Random byte generation failed.")
}
```

Comment (justindarc) [inline Shared/Bytes.swift]: @rnewman would there be any harm in just switching this code to use fatalError() regardless of the optimizations used?

Comment (rnewman) [inline Shared/Bytes.swift]: That would be fine: `precondition` and `fatalError` are mostly equivalent.

https://github.com/apple/swift/blob/adc54c8a4d13fbebfeb68244bac401ef2528d6d0/stdlib/public/core/Assert.swift#L170
https://github.com/apple/swift/blob/adc54c8a4d13fbebfeb68244bac401ef2528d6d0/stdlib/public/core/Assert.swift#L191

In a release build, `precondition` simply int-traps without printing. I don't think we care about that distinction.

**Pull request merged**